### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by katarighe**
1) **Removed** current_user method in ApplicationController, devise will provide us with one.

2) **Install and setup** devise.
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

3) **Modified** the views in devise for registration and login to match the wireframes from the Sneak Peek and my styling.

# Screenshots
**Login page**
image

**Signup Page**
image

**Email confirmation**
image

**Confirmation message**
image

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    @first_user = User.first
+  before_action :authenticate_user!
+  before_action :authenticate_user!
```
Comment (danmainah) [inline app/controllers/application_controller.rb]: - [ ] @katarighe and @jecode93 Great work adding authentication before loading the page, however, you have repeated this line of code twice, which is not a good practise.

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me, class: 'form-check-label' %>
+    </div>
+  <% end %>
+
+  <div class="actions my-2">
+    <%= f.submit "Log in", class:'btn btn-success' %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/sessions/new.html.erb]: - [ ] @Team Great work adding the login page, however the design slightly differs from the design given, the buttons should be the same width as the filed form, check the comparison below:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image |  image

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :password_confirmation, class: 'form-label'  %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password", class: 'form-control' %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up", class:'btn btn-info text-white' %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [ ] @Team Great work adding the registration page, however the design slightly differs from the design given, the buttons should be the same width as the filed form, check the comparison below:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image | image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @katarighe and @jecode93 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Setup the Devise correctly and ensure the user can Signup and Login successfully.
✔️  Modify the Signup and Login views to match the design given.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**